### DISCUSSION TITLE
图片里面的表头识别不出来

### DISCUSSION THREAD

**Discussion by Dkaaaaa** [RapidAI/TableStructureRec#84] [Q&A]
您好，我又来了，今天遇到一个图片里面有表头的，但是识别不出来（也有部分图片是可以识别出来的），想问一下是因为表头区域没有被划分为一个框导致吗，有没有什么参数可以解决呢？谢谢！

以下图片在识别之前已经经过角度纠正

不能识别表头的图片链接：https://patentimages.storage.googleapis.com/37/d2/c1/fe6296a7924d50/imgf000160_0001.png
可以识别表头的图片链接：https://patentimages.storage.googleapis.com/a9/ec/1b/c8e680af0dcf71/imgf000155_0001.png

**Comment by Joker1212** [comment]
有线表格本身其实原理是unet识别线框的，所以在线框外的部分，是识别不到的，有两个办法，一是直接依赖ocr识别结果把第一个当表头，二是使用我们的RapidLayout，对于这种文档，能版面区分表格部分和表头部分

**Reply by Joker1212** [reply]
RapidLayout执行也很快，不用担心资源情况，也是onnx模型，兼容性很好，准确度上应该也是开源最准的那个模型转换的，并提供多个版面识别模型

**Reply by Dkaaaaa** [reply]
您好，RapidLayout确实可以将表头识别出来，但是我后续用TableStructureRec进行识别还是会漏掉表头部分。我的TableStructureRec代码是您提供的代码。不知道是不是我使用有误呢

以下是我使用RapidLayout版面识别后的图片。
![image](https://github.com/user-attachments/assets/ef6c7d83-21c6-481a-8379-b21cef29bc26)

**Reply by Joker1212** [reply]
漏掉表头你是指漏掉第一行吗？我以为你说的表头是上面那个标题

**Reply by Dkaaaaa** [reply]
啊 对

**Reply by Dkaaaaa** [reply]
那应该叫表的题目。。抱歉

**Reply by Joker1212** [reply]
为啥我在线执行是有的，第一行也有 
<img width="1551" alt="image" src="https://github.com/user-attachments/assets/5f72c6bb-3186-4ccd-af7d-2eb889f54b70">

**Reply by Joker1212** [reply]
<img width="1609" alt="image" src="https://github.com/user-attachments/assets/e955bacb-afc1-4661-8f93-cacb47699140">
这个也行，就是多留了一点空列 ，理论上直接用layout的截图来识别的话，也能解决这个多余列的问题
<img width="1609" alt="image" src="https://github.com/user-attachments/assets/2ed5636c-d3cf-4ea1-ac6a-cbcaff37be18">

**Reply by Dkaaaaa** [reply]
怎么我的就不行。。
![image](https://github.com/user-attachments/assets/62f1002f-f1c4-4125-b0ed-bfb8e7458c7c)

**Reply by Dkaaaaa** [reply]
从这个box来看，他其实是有把这个表的题目框出来的，但是不知道为什么就没有识别出来
![image](https://github.com/user-attachments/assets/a6249e21-46de-452d-a10c-6e2b07f0fb4b)

**Reply by Joker1212** [reply]
ocr识别的是文字区域，有线表格是进行的线框检测，原理不一样，最后是做了box的包含关系生成的html结果，上面的标题部分可以用RapidLayout提取得到的吧，或者还有一个办法，你把图片稍微缩小一点，应该就可以了

**Reply by Joker1212** [reply]
流程可以是，layou输出标题+表格区域的截图，然后单独对表格截图进行表格识别，组合一下结果，也可以缩小一下图片试试能不能整个都用表格识别解决，或者格式固定的话，直接取ocr输出的第一个框就行了

**Reply by Dkaaaaa** [reply]
您好，基于您的建议，我尝试了（版面识别与否，不影响后续的识别）缩小图片尺寸后直接进行TableStructureRec识别提取内容，以下是我的缩小图片尺寸代码，如果我缩小50%，还是识别不了那张图片的表格的标题；如果我缩小80%，虽然是可以识别到标题了，但是分辨率被大大降低了，导致表格内容识别不出来了。想问一下有什么办法可以缩小图片尺寸又能保证清晰度呢？谢谢！

另外只取OCR第一个框也是一个不得不选的办法。

```python
from PIL import Image

# 打开图片
img = Image.open('/home/akesobio/tmp_ccy/siRNA/OCR/rotated_image.png')

# 获取原始图片的尺寸
original_width, original_height = img.size

# 设定缩小的比例（例如，缩小为原来的50%）
resize_ratio = 0.5

# 计算新尺寸
new_width = int(original_width * resize_ratio)
new_height = int(original_height * resize_ratio)

# 调整图片大小
resized_img = img.resize((new_width, new_height), Image.Resampling.LANCZOS)

# 保存缩小后的图片
resized_img.save('resized_image.jpg')


# 输出图片尺寸
print(f"图片原始尺寸: {original_width}x{original_height}")
print(f"图片缩小后尺寸: {new_width}x{new_height}")
```

**Reply by Joker1212** [reply]
我的意思是，layout既然能区分出标题和表格区域，那么对标题区域进行ocr就可以了吧，然后截取表格区域的box图片，进行表格识别，两个部分就隔离开了

**Reply by Joker1212** [reply]
版面识别的目的就是区分区域，进行不同的处理啊，版面识别会返回不同区域的box坐标，直接从原图截取，分别进行ocr和表格识别
